src/Microsoft.ML.TensorFlow/TensorflowTransform.cs

```diff
@@ -182,16 +166,16 @@ private static byte[] CheckFileAndRead(IHostEnvironment env, string modelFile)
         }
 
         public TensorFlowTransform(IHostEnvironment env, string modelFile, string[] inputs, string[] outputs) :
-            this(env, CheckFileAndRead(env, modelFile), inputs, outputs)
+            this(env, CheckFileAndRead(env, modelFile), inputs, outputs, modelFile)
         {
         }
 
-        private TensorFlowTransform(IHostEnvironment env, byte[] modelBytes, string[] inputs, string[] outputs)
+        private TensorFlowTransform(IHostEnvironment env, byte[] modelBytes, string[] inputs, string[] outputs, string modelFile = null)
         {
             Contracts.CheckValue(env, nameof(env));
             _host = env.Register(nameof(RegistrationName));
             _host.CheckValue(modelBytes, nameof(modelBytes));
-            Session = LoadTFSession(modelBytes);
+            Session = TensorFlowUtils.LoadTFSession(_host, modelBytes, modelFile);
```

Comment:
Abhishek is working on brining another way to load session, so I think it's better to extend CheckFileAndRead function and force it to return you Session instead of byte array. So private constructor would just  accept session.